adapters/socketio_mbed.c

```diff
@@ -281,7 +288,6 @@ int socketio_close(CONCRETE_IO_HANDLE socket_io, ON_IO_CLOSE_COMPLETE on_io_clos
         else
         {
             tcpsocketconnection_close(socket_io_instance->tcp_socket_connection);
-            socket_io_instance->tcp_socket_connection = NULL;
             socket_io_instance->io_state = IO_STATE_CLOSED;
```

Comment:
Why is this being deleted?

Reply:
correct. you can always reopen the connection. socketio_destroy() will destroy the socket connection.

tcpsocketconnection_create->tcpsocketconnection_connect->tcpsocketconnection_close->tcpsocketconnection_connect->tcpsocketconnection_close->tcpsocketconnection_destroy